MVCapplication/src/main/java/com/softserve/if072/mvcapp/controller/AnalyticsController.java

```diff
@@ -1,44 +1,96 @@
 package com.softserve.if072.mvcapp.controller;
 
-import com.softserve.if072.mvcapp.service.UserService;
-import org.apache.logging.log4j.LogManager;
-import org.apache.logging.log4j.Logger;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.beans.factory.annotation.Value;
+import com.softserve.if072.common.model.ProductStatistics;
+import com.softserve.if072.common.model.dto.AnalyticsProductDTO;
+import com.softserve.if072.mvcapp.service.AnalyticsService;
 import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.ExceptionHandler;
 import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.client.RestTemplate;
+import org.springframework.web.client.HttpClientErrorException;
+
+import javax.servlet.http.HttpSession;
+import java.util.List;
 
 /**
- * This class provides access to the analytics page.
+ * The CartController class handles requests for "/analytics" and renders appropriate view
+ *
+ * @author Igor Kryviuk
+ * @author Igor Parada
  */
 @Controller
 @RequestMapping("/analytics")
 public class AnalyticsController {
-    private static final Logger LOG = LogManager.getLogger(AnalyticsController.class);
+    private final AnalyticsService analyticsService;
 
-    @Value("${service.url.analytics}")
-    private String restAnalyticsURL;
+    public AnalyticsController(AnalyticsService analyticsService) {
+        this.analyticsService = analyticsService;
+    }
 
-    private UserService userService;
-    private RestTemplate restTemplate;
+    /**
+     * Handles requests for getting list of all user's products
+     *
+     * @param session - HttpSession object
+     * @param model   - a map that will be handed off to the view for rendering the data to the client
+     * @return - string with appropriate view name
+     */
+    @GetMapping()
+    public String getProducts(HttpSession session, Model model) {
+        ProductStatistics productStatistics = (ProductStatistics) session.getAttribute("productStatistics");
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "productStatistics" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1192)